curve_apps/contours/driver.py

```diff
@@ -49,60 +51,130 @@ def make_curve(self):
+            )
+
+            vertices, edges, values = ContoursDriver.get_contours(
+                grid, data, self.params.detection.contours
+            )
+
+            locations = vertices
+            if self.params.output.z_value:
+                locations = np.c_[locations, values]
+            else:
+                locations = set_vertices_height(locations, self.params.source.objects)
 
             curve = Curve.create(
                 self.workspace,
                 name=string_name(self.params.output.export_as),
                 vertices=locations,
-                cells=np.vstack(cells).astype("uint32"),
+                cells=edges,
                 parent=self.out_group,
             )
             curve.add_data(
-                {self.params.source.data.name: {"values": np.hstack(values)}}
+                {
+                    self.params.source.data.name: {
+                        "association": "VERTEX",
+                        "values": values,
+                    }
+                }
             )
 
             return curve
 
     @staticmethod
-    def get_contours(params: ContourParameters):
+    def get_contours(
+        grid: list[np.ndarray], data: np.ndarray, contour_list: list[float]
+    ) -> tuple[np.ndarray, np.ndarray, np.ndarray]:
         """
-        Calculate contour from source data.
+        Return vertices, edges, and values for contours.
 
-        :param params: Contour parameters
+        :param grid: list of x and y grids.
+        :param data: 2D array of data living in grid.
+        :param contour_list: list of contour values.
         """
 
-        entity = params.source.objects
-        data = params.source.data
-        locations = entity.locations
-        x, y = locations[:, :2].T
-        axis = plt.axes()
-        if isinstance(entity, Grid2D):
-            # TODO: Replace with scikit-image contour algorithm
-            contours = axis.contour(
-                x.reshape(entity.shape, order="F"),
-                y.reshape(entity.shape, order="F"),
-                data.values.reshape(entity.shape, order="F"),
-                levels=params.detection.contours,
-            )
-        else:
-            # TODO: Replace with scikit-image contour algorithm
-            contours = axis.tricontour(
-                x,
-                y,
-                data.values,
-                levels=params.detection.contours,
+        interp = ContoursDriver.image_to_grid(data, grid)
+        vertices, edges, values = [], [], []
+        v0 = 0
+        for contour in contour_list:
+
+            coords = measure.find_contours(data, contour, fully_connected="low")
+            if not coords:
+                continue
+
+            for coord in coords:
+                segment_vertices = interp(coord[:, 1], coord[:, 0])
+                nv = len(segment_vertices)
+                segment_edges = [[k, k + 1] for k in range(v0, v0 + (nv - 1))]
```

Comment:
```suggestion
                segment_edges = np.c_[np.arange(nv-1),  np.arange(1, nv)] + v0
```

More efficient than looping
![image](https://github.com/MiraGeoscience/curve-apps/assets/55204635/18420e19-c775-4ab3-aafc-c68014d53924)